packages/core/src/shared/fs/fs.ts

```diff
@@ -543,6 +543,34 @@ export class FileSystem {
         return this.#homeDir
     }
 
+    /**
+     * Gets the application cache folder for the current platform
+     *
+     * Follows the cache_dir convention outlined in https://crates.io/crates/dirs
+     */
+    getCacheDir(): string {
+        switch (process.platform) {
+            case 'darwin': {
+                return _path.join(this.getUserHomeDir(), 'Library/Caches')
+            }
+            case 'win32': {
+                const localAppData = process.env.LOCALAPPDATA
+                if (!localAppData) {
+                    throw new ToolkitError('LOCALAPPDATA environment variable not set', {
+                        code: 'LocalAppDataNotFound',
+                    })
+                }
+                return localAppData
+            }
+            case 'linux': {
+                return _path.join(this.getUserHomeDir(), '.cache')
+            }
+            default: {
+                throw new Error(`Unsupported platform: ${process.platform}. Expected 'darwin', 'win32', or 'linux'.`)
```

Comment:
for web should we return something? we support "home dir" on web: https://github.com/aws/aws-toolkit-vscode/blob/38842593a9e27dd00c750edfbc4b99aa7e4c1cd8/packages/core/src/shared/fs/fs.ts#L497-L502

Reply:
Just added it. It's probably useful to have if I need it for web support in the next couple weeks